packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -184,7 +184,7 @@ export function createOdspNetworkError(
                 if (redirectLocation !== undefined) {
                     const propsWithRedirectLocation = { ...driverProps, redirectLocation };
                     error = new NonRetryableError(
```

Comment:
What do you think about this - Update `class LocationRedirectionError` (see below, and update the interface too) and then return that class here.  Then simply modify the error object to add on `redirectUrl` in `epochTracker`.  This way there's only ever one error object, and the errorType and class are coupled which is preferred.

```typescript
export class LocationRedirectionError extends LoggingError implements ILocationRedirectionError, IFluidErrorBase {
    readonly errorType = DriverErrorType.locationRedirection;
    readonly canRetry = false;

    redirectUrl?: IResolvedUrl;

    constructor(
        message: string,
        readonly redirectLocation: string,
        props: DriverErrorTelemetryProps,
    ) {
        super(message, props);
    }
}
```

Reply:
Another idea would be to have an ODSP-specific implementation the extended the driver base one, and it has the logic for `redirectLocation` -> `redirectResolvedUrl`.  It would take the driver base one as a constructor param and its implementation would just forward all properties to the inner base one?